src/ast/checks.rs

```diff
@@ -985,6 +985,44 @@ pub fn unnecessary_collection_call(
     None
 }
 
+pub fn unnecessary_subscript_reversal(expr: &Expr, func: &Expr, args: &[Expr]) -> Option<Check> {
+    if let Some(first_arg) = args.first() {
+        if let ExprKind::Name { id, .. } = &func.node {
+            if id == "set" || id == "sorted" || id == "reversed" {
+                if let ExprKind::Subscript { slice, .. } = &first_arg.node {
+                    if let ExprKind::Slice { lower, upper, step } = &slice.node {
+                        if lower.is_none() && upper.is_none() {
+                            if let Some(step) = step {
+                                if let ExprKind::UnaryOp {
+                                    op: Unaryop::USub,
+                                    operand,
+                                } = &step.node
+                                {
+                                    if let ExprKind::Constant {
+                                        value: Constant::Int(val),
+                                        ..
+                                    } = &operand.node
+                                    {
+                                        if val.to_string() == "1" {
```

Comment:
Tried `val == 1` first, but got this error:

<img width="754" alt="image" src="https://user-images.githubusercontent.com/17039389/194740275-1cb03619-7e4c-492b-bd81-ab5fdc8d5268.png">

Reply:
Would it be safer to create BigInt(1) and compare them as BigInts?

Reply:
Yeah, that would be safer. Fixed :)